cmd/soroban-rpc/internal/preflight/pool.go

```diff
@@ -65,25 +113,49 @@ func (pwp *PreflightWorkerPool) Close() {
 
 var PreflightQueueFullErr = errors.New("preflight queue full")
 
+type metricsLedgerEntryWrapper struct {
+	db.LedgerEntryReadTx
+	totalDurationMs uint64
+}
+
+func (m *metricsLedgerEntryWrapper) GetLedgerEntry(key xdr.LedgerKey) (bool, xdr.LedgerEntry, error) {
+	startTime := time.Now()
+	ok, entry, err := m.LedgerEntryReadTx.GetLedgerEntry(key)
+	atomic.AddUint64(&m.totalDurationMs, uint64(time.Since(startTime).Milliseconds()))
+	return ok, entry, err
+}
+
 func (pwp *PreflightWorkerPool) GetPreflight(ctx context.Context, readTx db.LedgerEntryReadTx, sourceAccount xdr.AccountId, op xdr.InvokeHostFunctionOp) (Preflight, error) {
 	if pwp.isClosed.Load() {
 		return Preflight{}, errors.New("preflight worker pool is closed")
 	}
+	wrappedTx := metricsLedgerEntryWrapper{
+		LedgerEntryReadTx: readTx,
+		totalDurationMs:   0,
+	}
 	params := PreflightParameters{
 		Logger:             pwp.logger,
 		SourceAccount:      sourceAccount,
 		InvokeHostFunction: op,
 		NetworkPassphrase:  pwp.networkPassphrase,
-		LedgerEntryReadTx:  readTx,
+		LedgerEntryReadTx:  &wrappedTx,
 	}
 	resultC := make(chan workerResult)
 	select {
 	case pwp.requestChan <- workerRequest{ctx, params, resultC}:
 		result := <-resultC
+		status := "ok"
+		if result.err != nil {
+			status = "error"
+		}
+		pwp.durationMetric.With(
+			prometheus.Labels{"type": "db", "status": status},
+		).Observe(float64(wrappedTx.totalDurationMs * 1000))
```

Comment:
Which unit did you intend to meter here ? It's not seconds here but rather microsecond. Was that your intent ?

Reply:
assuming you'll call the metric `request_ledger_get_duration_seconds`, then you'd need to divide by 1000. obviously,  do so outside of the float64 conversion